contracts/token/ERC721/ERC721.sol

```diff
@@ -220,16 +217,22 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
         return _ownerOf(tokenId) != address(0);
     }
 
+    /**
+     * @dev Returns the approved address for `tokenId`. Returns 0 if `tokenId` is not minted.
+     */
+    function _getApproved(uint256 tokenId) internal view virtual returns (address) {
+        return _tokenApprovals[tokenId];
+    }
+
     /**
      * @dev Returns whether `spender` is allowed to manage `tokenId`.
-     *
-     * Requirements:
-     *
-     * - `tokenId` must exist.
      */
     function _isApprovedOrOwner(address spender, uint256 tokenId) internal view virtual returns (bool) {
-        address owner = ownerOf(tokenId);
-        return (spender == owner || isApprovedForAll(owner, spender) || getApproved(tokenId) == spender);
+        address owner = _ownerOf(tokenId);
+        return (owner != address(0) &&
+            (spender == owner ||
+                isApprovedForAll(owner, spender) ||
+                (_getApproved(tokenId) == spender && spender != address(0))));
```

Comment:
No actionnable here, but for the record I'm not a big fan of the linter's style.